packages/sitecore-jss-nextjs/src/sharedTypes/component-props.ts

```diff
@@ -0,0 +1,30 @@
+import { GetServerSidePropsContext, GetStaticPropsContext } from "next";
+import {ComponentRendering, LayoutServiceData} from '@sitecore-jss/sitecore-jss';
+
+/**
+ * Shape of component props storage
+ */
+export type ComponentProps = {
+  [componentName: string]: unknown;
```

Comment:
Given my comment above about needing to handle multiple components with the same name - this will likely need to change to a uid key.

Reply:
Changed